### PR TITLE
Support MUI v5 (and upgrade to React v17).

### LINKED ISSUE #305: Support for Material-UI version 5
### Describe the problem related to this feature request

Currently, this project does not work with Material-UI version 5, however, the changes required to do so are minimal. It seems that the only breaking change is that makeStyles has been moved from @material-ui/core/styles to @material-ui/styles.

### Describe the solution you'd like

From my limited testing, accounting for this requires the following changes:  

1. A 4.x branch must be created to host the changes
2. The README.md file must be updated on each support branch to reflect the changes
3. DropzoneArea.md must change the makeStyles import line to use the new import
3. The package.json file must upgrade its Material-UI dependencies to ^5
4. The package.json file must add a new dependency of @material-ui/styles
6. Any instances of makeStyles or withStyles must change their imports to @material-ui/styles

### Describe alternatives you've considered

If I want to use Material-UI v5 with this package, the only option I have right now is to use patch-package. In case it's of use, this is the patch I made to get my project to compile:  

> --- a/node_modules/material-ui-dropzone/dist/index.es.js
+++ b/node_modules/material-ui-dropzone/dist/index.es.js
@@ -12,7 +12,7 @@ import PropTypes from 'prop-types';
 import { createElement, Fragment, PureComponent } from 'react';
 import Snackbar from '@material-ui/core/Snackbar';
 import Typography from '@material-ui/core/Typography';
-import { withStyles } from '@material-ui/core/styles';
+import { withStyles } from '@material-ui/styles';
 import AttachFileIcon from '@material-ui/icons/AttachFile';
 import CloudUploadIcon from '@material-ui/icons/CloudUpload';
 import clsx from 'clsx';

### PR DESCRIPTION
**Pull request by sah**
## Description

Support MUI v5 (and upgrade to React v17, as required by MUI). This just required updating package.json and changing package imports.

- Fixes #305

## Type of change

- Breaking change (fix or feature that would cause existing functionality to not work as expected)

## How Has This Been Tested

Manually tested, and I've been using this on a production website for several months.

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: src/components/DropzoneArea.md [hunk 1]
```diff
@@ -78,7 +78,7 @@ const file = new File(["foo"], "foo.txt", {
 Chips use the Grid system as well, so you can customize the way they appears and benefit from the Material-UI grid customizations
 
 ```jsx
-import { createStyles, makeStyles } from '@material-ui/core/styles';
+import { createStyles, makeStyles } from '@material-ui/styles';
```
Comment (baurt) [inline src/components/DropzoneArea.md:81]: I think this should be `import { makeStyles, createStyles } from '@mui/styles';`

**Comment by baurt** [comment]
Should `@mui/styles` be added to package.json? Otherwise, it requires a separate install.

**Comment by sah** [comment]
> Should `@mui/styles` be added to package.json? Otherwise, it requires a separate install.

I agree it should. Added.

**Comment by LucasBassetti** [comment]
Hey! Any predictions to merging this? It will be good to use this package with v5

**Comment by amireallyadev** [comment]
I am waiting for this too. I can not migrate to MUI 5 due to this.

**Comment by LucasBassetti** [comment]
@Yuvaleros could you please merge this and create a new release?

**Comment by baurt** [comment]
> @Yuvaleros could you please merge this and create a new release?

There is a [fork](https://github.com/lifeeric/material-ui-dropzone) that has merged equivalent changes and fixed an issue with the snackbar. It's published on npm as [react-mui-dropzone](https://www.npmjs.com/package/react-mui-dropzone)

**Comment by puriasafarihesari** [comment]
@baurt, I have been trying out npm package you're referring to for the past couple of days and I'm still unable to override the styles with the various namespaces such as MuiDropezoneArea. Do you have a working example of that by any chance?

**Comment by baurt** [comment]
@puriasafarihesari  No. I haven't tried overriding the styles. I use the package and styles as is.

**Comment by cannontechnology** [comment]
@Yuvaleros can you please merge some of these pull requests or add someone else as a maintainer?

**Comment by LucasBassetti** [comment]
@Yuvaleros thanks! 🤩  Will you publish a new npm release with v5 support?

**Comment by cannontechnology** [comment]
Thanks, @Yuvaleros!  Can you publish a new npm release?

**Pull request merged**